### PR TITLE
feat: add automated unit testing

### PR DESCRIPTION
**Pull request by bolny**
# Summary
This PR adds comprehensive unit testing for the warning selection logic and refactors the existing Python code to improve testability, readability, and maintainability.

# Changes

## Added
- GitHub Actions workflow for automated testing on PRs and pushes to main
- Extensive unit tests covering:
  - Core warning selection logic
  - Timezone handling (PT/UTC edge cases)
  - Special handling for wildfire smoke warnings
  - Warning expiry verification
- Type hints to improve code readability and IDE support
- Better error handling for file processing

## Refactored
- Restructured [`frontend/construct-lists.py`](frontend/construct-lists.py ) to separate concerns:
  - Extracted header parsing logic to a dedicated function
  - Created standalone selection function with improved testability
  - Added main function to organize high-level flow
- Updated dependencies to include missing [`pytz`](frontend/.venv/lib/python3.13/site-packages/pytz/__init__.py ) package

## Fixed
- Prevented crash when `QUARTO_PROJECT_INPUT_FILES` environment variable is not set
- Added testing for timezone edge case when warnings incorrectly expire early due to UTC vs PT timezone differences

# Testing
All unit tests pass and cover both regular cases and edge cases, particularly the timezone-dependent expiry logic that ensures warnings remain active until midnight Pacific Time rather than UTC.

### CODE REVIEW

File: frontend/construct-lists.py [hunk 1]
```diff
@@ -19,66 +20,138 @@
...
+    try:
+        with open(file_path, "r") as file:
             contents = file.read()
             match = HEADER_REGEX.search(contents)
             if match:
                 doc_preamble = match.group(1)
                 parsed_header = yaml.safe_load(doc_preamble)
-
-                # what goes in the generated yaml
+
```
[the remark this answers is not in the document]
Comment (bolny) [reply frontend/construct-lists.py:52]: Remove extraneous whitespace.

**Review by dawnbcgov — COMMENTED** [review]
Left some small comments inline

File: frontend/test_construct_lists.py
[the remark this answers is not in the document]
Comment (bolny) [reply frontend/test_construct_lists.py:1]: It depends on the project. I think that there is so little logic that needs to be tested here that it doesn't necessarily make the code structure more easy to navigate. I don't have strong feelings either way.

File: frontend/construct-lists.py
[the remark this answers is not in the document]
Comment (bolny) [reply frontend/construct-lists.py:1]: Yeah that's probably wise. I think I'd like to do that as a separate pull request though to keep the scope as narrow as possible.

File: .github/workflows/python-tests.yml
[the remark this answers is not in the document]
Comment (bolny) [reply .github/workflows/python-tests.yml:1]: What you are describing are integration or end to end tests. Yes, we would have a different workflow for those. In this case the python unit test runner automatically discovers all python unit tests. The action is very quick (completes in under 20s, and there aren't any costs associated with it), so I'm not too worried about optimization here.

The more pressing issue, as I see it, is that we really want to make passing tests a required check to block merging of new code that breaks the tests. I'm not aware of a way with GitHub to make this requirement conditional, so it will need to run with every pull request.

**Review by dawnbcgov — APPROVED** [review]
Thanks for the responses. It sounds like some potential future work but no changes to this PR so I'll leave to you to merge!

File: frontend/construct-lists.py
[the remark this answers is not in the document]
Comment (bolny) [reply frontend/construct-lists.py:1]: I created https://github.com/bcgov/aqwarnings/issues/121 to track this. I might not get to it today, but it should be quick, and then I'll move on to https://github.com/bcgov/aqwarnings/issues/79
